node/network/approval-distribution/src/lib.rs

```diff
@@ -1758,3 +1733,75 @@ impl<Context> ApprovalDistribution {
 		SpawnedSubsystem { name: "approval-distribution-subsystem", future }
 	}
 }
+
+/// Ensures the batch size is always at least 1 element.
+const fn ensure_size_not_zero(size: usize) -> usize {
+	if 0 == size {
+		panic!("Size must be at least 1");
```

Comment:
```suggestion
		panic!("Batch size must be at least 1 (MAX_NOTIFICATION_SIZE constant is too low: {})", MAX_NOTIFICATION_SIZE);
```

Reply:
done but without the extra formatting - `cannot call non-const formatting macro in constant functions`. Decided not to use const_format as it would bring and additional dependency and the error is very obvious any way.